libraries/hts221_driver/hts221.c

```diff
@@ -0,0 +1,301 @@
+    hum = ((int16_t)htc221_sensor.calib.H0_rH_x2) / 2.0; // Remove x2 multiple.
+    _humidity = (int16_t)((hum + h_temp));               // Provide signed % measurement unit.
+
+    // Limit output range according to HTS221 datasheet.
+    if (_humidity >= HUMID_MAX)
+    {
+        _humidity = HUMID_MAX;
+    }
+    else if (_humidity <= HUMID_MIN)
+    {
+        _humidity = HUMID_MIN;
+    }
+
+    return (_humidity);
+}
+
+float hts221_temperature_get(void)
+{
+    int16_t t_out;
+    uint8_t temp_h;
+    uint8_t temp_l;
+
+    float deg = 0.0f;
+    float t_temp = 0.0f;
+    float _temperature = 0.0f;
+
+    htc221_readI2C(TEMP_OUT_L_REG, &temp_l, 1);
+    htc221_readI2C(TEMP_OUT_H_REG, &temp_h, 1);
+
+    t_out = ((uint16_t)temp_h << 8) + temp_l;
+    // T0_degC_x8.
+    deg = ((int16_t)(htc221_sensor.calib.T1_degC_x8) - (int16_t)(htc221_sensor.calib.T0_degC_x8)) / 8.0f; // remove x8 multiple
+    // Calculate Temperature in decimal of grade centigrades i.e. 15.0 = 150.
+    t_temp = (((int16_t)t_out - (int16_t)htc221_sensor.calib.T0_OUT) * deg) / ((int16_t)htc221_sensor.calib.T1_OUT - (int16_t)htc221_sensor.calib.T0_OUT);
+    deg = ((int16_t)htc221_sensor.calib.T0_degC_x8) / 8.0; // Remove x8 multiple.
+    _temperature = deg + t_temp;                           // Provide signed celsius measurement unit.
+
+    return _temperature;
+}
+
+int32_t hts221_calib_get(hts221_calib_t *p_calib)
+{
+    uint8_t calib_raw[CALIBRATION_REGS_NUM];
+    int32_t err;
+
+    err = htc221_readI2C(CALIBRATION_REGS, calib_raw, CALIBRATION_REGS_NUM);
+
+    p_calib->H0_rH_x2 = calib_raw[0];
+    p_calib->H1_rH_x2 = calib_raw[1];
+    p_calib->T0_degC_x8 = (uint16_t)calib_raw[2] + ((uint16_t)(calib_raw[5] & 0x03) << 8);
+    p_calib->T1_degC_x8 = (uint16_t)calib_raw[3] + ((uint16_t)((calib_raw[5] >> 2) & 0x03) << 8);
+    p_calib->H0_T0_OUT = (int16_t)calib_raw[6] + ((int16_t)calib_raw[7] << 8);
+    p_calib->H1_T0_OUT = (int16_t)calib_raw[10] + ((int16_t)calib_raw[11] << 8);
+    p_calib->T0_OUT = (int16_t)calib_raw[12] + ((int16_t)calib_raw[13] << 8);
+    p_calib->T1_OUT = (int16_t)calib_raw[14] + ((int16_t)calib_raw[15] << 8);
+
+    return err;
+}
+
+int32_t hts221_one_shot(void)
+{
+    int32_t err;
+    uint8_t reg_val;
+
+    err = htc221_readI2C(CTRL_REG2, &reg_val, 1);
+    if (reg_val & CTRL_REG2_ONE_SHOT_Msk)
+    {
+        return 0;
+    }
+
+    reg_val |= CTRL_REG2_ONE_SHOT_Msk;
+
+    err = htc221_writeI2C(CTRL_REG2, &reg_val, 1);
+    return err;
+}
+
+int32_t hts221_cfg_set(hts221_cfg_t *p_cfg)
+{
+    int32_t err;
+
+    if ((p_cfg->av_conf & AV_CONF_RESERVED_BITS) ||
+        (p_cfg->ctrl_reg1 & CTRL_REG1_RESERVED_BITS) ||
+        (p_cfg->ctrl_reg2 & CTRL_REG2_RESERVED_BITS) ||
+        (p_cfg->ctrl_reg3 & CTRL_REG3_RESERVED_BITS))
+    {
+        return -1;
+    }
+
+    err = htc221_writeI2C(AV_CONF_REG, &(p_cfg->av_conf), 1);
```

Comment:
the error err is overwritten in the following lines. Maybe you want to use `|=`?
Maybe one could set err to be an uint32_t?

Reply:
Same like above